### DISCUSSION TITLE
Use of getconnect

### DISCUSSION THREAD

**Discussion by FeliceT57** [jonataslaw/getx#2991] [Q&A]
I want to know why this function is not correct:
Future<Map<String, dynamic>?> fetchData() async {
    GetConnect connect = GetConnect();
    final Response response = await connect.get('192.168.1.21/test.php');
    if (response.statusCode == 200) {
      return {
        'status': true,
        'response': response.body,
      };
    } else {
      return {'status': false, 'error': response.body};
    }
  }
 
The response.statuscode is correct (200) but the contents of response.body is this:
"<!DOCTYPE html> etc.....

**Comment by FeliceT57** [comment]
I rewrite my code:
import 'package:flutter/material.dart';
import 'package:get/get.dart';

void main() {
  runApp(const MainApp());
}

class MainApp extends StatefulWidget {
  const MainApp({super.key});

  @override
  State<MainApp> createState() => _MainAppState();
}

class _MainAppState extends State<MainApp> {
  Service s = Service();
  @override
  void initState() {
    s.fetchData();
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      home: Scaffold(
        body: Center(
          child: Text("Ciao Mondo!"),
        ),
      ),
    );
  }
}

class Service extends GetConnect {
  @override
  void onInit() {
    httpClient.baseUrl = 'http://127.0.0.1/';
    httpClient.timeout = const Duration(seconds: 8);
    super.onInit();
  }

  Future<Map<String, dynamic>?> fetchData() async {
    final response = await get(
      'test.php',
    );
    if (response.statusCode == 200) {
      print(response.body);
      return {
        'status': true,
        'response': response.body,
      };
    } else {
      return {'status': false, 'error': response.body};
    }
  }
}

like this the php script is not executed; the response.statuscode is equal 200 and the response.body is about "<!DOCTYPE html>
<html>
<head>
  <!-- etc... (it is the contents of the index.html)

If i change this:
final response = await get(
      'http://127.0.0.1/test.php',
the php script is executed the response is null.

Here the php script:
<?php
$response = [
  'data' => 'Ciao mondo!',
];
header('HTTP/1.1 200 OK');
echo json_encode($response);
?>

**Comment by inyong1** [comment]
Check yout test.php
Makesure the test.php send correct http status code depending on it's logic, success or not

**Reply by FeliceT57** [reply]
Thank'you for your interest. I found my error: Service s = Service(); is my error. Correct is: Service s = Get.put(Service());
